doc/man3/SSL_shutdown.pod

```diff
@@ -347,6 +347,25 @@ If an application calls SSL_shutdown_ex() with B<SSL_SHUTDOWN_FLAG_RAPID>, an
 application can subsequently change its mind about performing a rapid shutdown
 by making a subsequent call to SSL_shutdown_ex() without the flag set.
 
+=head2 Peer-Initiated Shutdown
+
+In some cases, an application may wish to wait for a shutdown initiated by the
+peer rather than triggered locally. To do this, call SSL_shutdown_ex() with
+I<SSL_SHUTDOWN_FLAG_WAIT_PEER> specified in I<flags>. In blocking mode, this
+waits until the peer initiates a shutdown or the connection otherwise becomes
+terminated for another reason. In nonblocking mode it exits immediately with
+either success or failure depending on whether a shutdown has occurred.
```

Comment:
My understanding is that in this case we don't wait for the streams to flush (should we?) and `SSL_SHUTDOWN_FLAG_NO_STREAM_FLUSH` has no effect. Shouldn't we say something about this in the docs?

Reply:
Added.